src/app/modules/auth/auth.middlewares.ts

```diff
@@ -53,3 +57,34 @@ export const denyRpSpStudentEmails: ControllerHandler<
         .json({ message: 'User not found' }),
     )
 }
+
+/**
+ * Logs all admin actions which change database state (i.e. non-GET requests)
+ * @returns next
+ */
+export const logAdminAction: ControllerHandler<{ formId: string }> = async (
```

Comment:
I'm not sure it's safe to always assume that `formId` will always be a route parameter. that may be the case currently (I'm not even sure) but idt we should couple the implementation to this assumption.

Reply:
as discussed, this makes sense to log form-specific actions!